File: backend/src/main/java/de/ckollmeier/burgerexpress/backend/dto/OrderableItemOutputDTO.java
package de.ckollmeier.burgerexpress.backend.dto;

import java.util.List;
import java.util.Map;

public record OrderableItemOutputDTO(
        String id,
        String name,
        String oldPrice,
        String price,
        String type,
        Map<String, List<String>> imageUrls,
        List<String> descriptionForDisplay,
        List<String> descriptionForCart
) {
}

File: frontend/src/types/OrderableItemOutputDTO.ts
export type OrderableItemOutputDTO = {
    id: string;
    name: string;
    oldPrice: string | null;
    price: string;
    type: string;
    imageUrls: Record<string, string[]>;
    descriptionForDisplay: string[];
    descriptionForCart: string[];
};

export function isOrderableItemOutputDTO(item: unknown): item is OrderableItemOutputDTO {
    return (
        typeof item === 'object' &&
        item !== null &&
        'id' in item &&
        'name' in item &&
        'oldPrice' in item &&
        'price' in item &&
        'type' in item &&
        'imageUrls' in item &&
        'descriptionForDisplay' in item &&
        'descriptionForCart' in item &&
        typeof item.id === 'string' &&
        typeof item.name === 'string' &&
        (item.oldPrice === null || typeof item.oldPrice === 'string') &&
        typeof item.price === 'string' &&
        typeof item.type === 'string' &&
        typeof item.imageUrls === 'object' &&
        Array.isArray(item.descriptionForDisplay) &&
        Array.isArray(item.descriptionForCart)
    );
}